src/engine/Server.cpp

```diff
@@ -128,7 +122,7 @@ void Server::run(const string& indexBaseName, bool useText, bool usePatterns,
     // in the catch block, hence the workaround with the `exceptionErrorMsg`.
     std::optional<std::string> exceptionErrorMsg;
     try {
-      co_await process(std::move(request), sendWithAccessControlHeaders);
+      co_await process(request, sendWithAccessControlHeaders);
```

Comment:
This function takes a const-reference as argument so the move doesn't make any sense here. Also there's a case where the un-moved object is required further down the control flow

Reply:
Good catch!